### PR TITLE
Remove prefix check on custom/third-party metrics.

### PR DESCRIPTION
**Pull request by igorpeshansky**
Also change label type check into a warning and make the metric_type type check effective.

### CODE REVIEW

File: src/write_gcm.c [hunk 1]
```diff
@@ -3019,25 +3020,16 @@ static int wg_json_CreateTimeSeries(
...
-        }
-        const char *pref = custom_metric_prefix;
-        if (strncmp(entry->value.value_text, pref, strlen(pref)) != 0) {
-          ERROR("write_gcm: plugin: %s, plugin_type: %s, metric_type: %s, "
-                "type_instance: %s metric type %s is not a custom metric "
-                "(should start with '%s').",
-                head->key.plugin, head->key.plugin_instance, head->key.type,
-                head->key.type_instance, entry->value.value_text, pref);
-          continue;
+          continue next_payload;
```
Comment (igorpeshansky) [inline src/write_gcm.c]: Yes. That's what the "make the metric_type check effective" refers to in the PR description.

File: src/write_gcm.c [hunk 2]
```diff
@@ -3019,25 +3020,16 @@ static int wg_json_CreateTimeSeries(
...
       }
       const char *key_pref = custom_metric_label_prefix;
       if (strncmp(entry->key, key_pref, strlen(key_pref)) == 0) {
         if (entry->value.value_type != wg_typed_value_string) {
-          ERROR("write_gcm: plugin: %s, plugin_type: %s, metric_type: %s, "
-                "type_instance: %s metric label %s is not a string.",
-                head->key.plugin, head->key.plugin_instance, head->key.type,
-                head->key.type_instance, entry->key);
+          WARNING("write_gcm: plugin: %s, plugin_type: %s, metric_type: %s, "
+                  "type_instance: %s metric label %s is not a string.",
```
Comment (dhrupadb) [inline src/write_gcm.c]: As discussed offline, would it make sense to mention something here about requiring user defined Metric Descriptors in order for the ingestion to succeed? As the error from this situation would happen on the API side.

**Comment by sparkprime** [comment]
How come you're removing the prefix check then, is it implicitly added now or something?

File: src/write_gcm.c (hunk 2 above)
Comment (igorpeshansky) [inline src/write_gcm.c]: The API will actually create metric descriptors with string labels, and then attempt to write the time series with this label into that metric descriptor, which will fail. The error returned from the API will be something like "label <foo> is not a string". In combination with this message, it should direct the users to the source of the problem.

**Comment by igorpeshansky** [comment]
@sparkprime The prefix check was wrong -- you can send any publicly writable metric through this mechanism; they don't have to start with `custom.googleapis.com`.

**Review by dhrupadb — APPROVED** [review]
LGTM

**Pull request merged**